repository/src/main/java/org/apache/atlas/repository/store/graph/AtlasTypeDefGraphStore.java

```diff
@@ -381,6 +381,12 @@ public AtlasTypesDef updateTypesDef(AtlasTypesDef typesDef) throws AtlasBaseExce
 
         AtlasTypesDef ret = updateGraphStore(typesDef, ttr);
 
+        try {
+            ttr.updateTypesWithNoRefResolve(ret);
```

Comment:
Call updateTypes(), instead of updateTypesWithNoRefResolve() - "*WithNoRefResolve()" calls will not run resolveReferences() hence will leave type objects with no reference to other types (for example: an entity-type will not have its references set for other entity-types references by it).